### PR TITLE
Packaging and test framework

### PR DESCRIPTION
**Pull request by pshriwise**
This adds packaging info to improve ease of installation and use for the plotting tool.

It also includes the beginning of a test framework for the plotter. Thanks to an extension of `pytest`, [pytest-qt](https://pytest-qt.readthedocs.io/en/latest/), we should be able to simulate different events in the app and generate outputs to ensure that code changes aren't breaking basic functionality. The `qtbot` fixture captures widgets and handles events appropriately as if a proper `QApplication` were running while reporting exceptions/failures to pytest.

Many tests need to be added and the framework could be improved, but I felt this was a good incremental step to continue from for future testing work.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@pytest.fixture(scope='module', autouse=True)
+def setup_regression_test(request):
+    # Change to test directory
+    olddir = request.fspath.dirpath().chdir()
+    try:
+        yield
+    finally:
+        # some cleanup
+        if Path("./plot_settings.pkl").exists():
+            os.remove("plot_settings.pkl")
```
Comment (paulromano) [inline tests/conftest.py]:
```suggestion
        plot_settings = Path('plot_settings.pkl')
        if plot_settings.exists():
            plot_settings.unlink()
```

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+    ],
+
+    # Dependencies
+    'python_requires': '>=3.5',
+    'install_requires': [
+        'openmc>=0.11.0', 'numpy', 'matplotlib', 'PySide2'
```
Comment (paulromano) [inline setup.py]: We should put the test requirements (pytest, pytest-qt) somewhere in here. For OpenMC proper, we put it in a section in `extras_require` -- up to you if you want to adopt that approach.

Comment (pshriwise) [reply setup.py]: Good call. I'm good with that approach. Updating now!

**Comment by pshriwise** [comment]
Re: implicit namespace - Yeah I guess my initial inclination would be to move the current plotter module to something like `xs_plotter` and then we could claim that namespace for this tool instead.

**Pull request merged**